### ISSUE TITLE
Thrown `ServerError` are not catched by `ErrorBoundary`

### ISSUE THREAD

**Issue by binajmen** [solidjs/solid-start#596]
I managed to reduce my current MVP to a reduced part (can't share the whole thing as it is for a private company) while reproducing the issue I'm encountering.

In the [following code](https://github.com/binajmen/solid-server-error/blob/8ee27dba3cc0305b2954a371276709c4f0260073/src/routes/%5Blang%5D/%5BtestId%5D/instructions.tsx#L9-L21), with an empty database, `findUniqueOrThrow()` will throw. This is the purpose of this example. It should be catched by either the local `ErrorBoundary` or the one defined in `root.tsx`

```ts
export function routeData({ params }: RouteDataArgs) {
  return createServerData$(
    async (id, { request }) => {
      // throw new ServerError("fail"); <-- will lead to same issue
      const test = await prisma.test.findUniqueOrThrow({
        where: { id },
      });

      return test;
    },
    { key: () => params.testId }
  );
}
```
```ts
NotFoundError [PrismaClientKnownRequestError]: No Test found
```
But in production the error is not catched, therefore kills the node process.

You can reproduce with:

1. git clone https://github.com/binajmen/solid-server-error.git
2. npm install
3. npm run build
4. npm run start
5. Visit http://localhost:3000/:lang/:testId/instructions

```ts
➜  temp git clone https://github.com/binajmen/solid-server-error.git
➜  temp cd solid-server-error
➜  solid-server-error git:(main) npm install
➜  solid-server-error git:(main) npm run build
➜  solid-server-error git:(main) npm run start

> start
> solid-start start

 solid-start start
 version  0.2.8
 adapter  node


  ➜  Page Routes:
     ┌─ http://localhost:3000/*404
     └─ http://localhost:3000/:lang/:testId/instructions

  ➜  API Routes:
     None! 👻

🔌 setting up prisma client to file:///dev.db
Listening on port 3000
node:internal/process/promises:288
            triggerUncaughtException(err, true /* fromPromise */);
            ^

NotFoundError [PrismaClientKnownRequestError]: No Test found
    at /Users/Projects/temp/solid-server-error/node_modules/@prisma/client/runtime/index.js:34217:15
    at async /Users/Projects/temp/solid-server-error/node_modules/@prisma/client/runtime/index.js:34215:20
    at async Object.$$serverHandler0 (file:///Users/Projects/temp/solid-server-error/dist/server.js:23973:16)
    at async execute (file:///Users/Projects/temp/solid-server-error/dist/server.js:23747:17)
    at async resourceFetcher (file:///Users/Projects/temp/solid-server-error/dist/server.js:23371:22) {
  code: 'P2025',
  clientVersion: '4.8.0',
  meta: undefined,
  batchRequestIdx: undefined
}

Node.js v18.8.0
```

**Comment by sgwong** [comment]
Probably the fix in #366 also fixes this issue? node server no longer crashes and shows 404 Page not found after the fixed.

**Comment by nksaraf** [comment]
Will look into this, was away for holidays.

**Comment by sgwong** [comment]
I should be fixed with changes from issue #366. After apply that changes it able caught the error.

**Comment by ryansolid** [comment]
In setting up for SolidStarts next Beta Phase built on Nitro and Vinxi we are closing all PRs/Issues that will not be merged due to the system changing. If you feel your issue was closed by mistake. Feel free to re-open it after updating/testing against 0.4.x release. Thank you for your patience.  See https://github.com/solidjs/solid-start/pull/1139 for more details.
